versions/development/SPEC.md

```diff
@@ -1218,81 +1270,316 @@ $runtime = 'runtime' $ws* '{' ($ws* $runtime_kv $ws*)* '}'
+
+Container source locations should use the syntax defined by the individual container repository. For example an image defined as `ubuntu:latest` would conventionally refer a docker image living on `DockerHub`, while an image defined as `quay.io/bitnami/python` would refer to a `quay.io` repository. 
+
+The supported attribute types are:
+
+* `String`: A single container location
+* `Array[String]`: A list of container entries. **Note**: the ordering of the list does not imply any implicit preference or ordering of the containers. All images are expected to be the same, and therefore any choice would be equally valid.
+
+**Single Image**
 ```wdl
-task test {
-  input {
-    String ubuntu_version
-  }
-  command {
-    python script.py
+task single_image_test {
+  #....
+  runtime {
+    container: "ubuntu:latest"
   }
+```
+
+**Multiple Images**
+```wdl
+task multiple_image_test {
+  #.....
   runtime {
-    docker: "ubuntu:" + ubuntu_version
+    container: ["ubuntu:latest", "gcr.io/standard-images/ubuntu:latest"]
   }
 }
 ```
 
-Most key/value pairs are arbitrary.  However, the following keys have recommended conventions:
 
-#### docker
+#### cpu (**Required**)
 
-Location of a Docker image for which this task ought to be run.  This can have a format like `ubuntu:latest` or `broadinstitute/scala-baseimage` in which case it should be interpreted as an image on DockerHub (i.e. it is valid to use in a `docker pull` command).
+The `cpu` key defines the _minimum_ CPU required for this task, which must be available prior to the engine starting execution. The engine does not need to provide the exact amount of CPU requested (depending on the underlying infrastructure restrictions), however it may ONLY provision more CPU than requested and not less. For example if the wdl requested `cpu: 0.5`, but only discrete values were supported, then the engine might choose to provision `1.0` cpu instead.  Values are expected to be a `Float` or an `Int`
 
 ```wdl
-task docker_test {
-  input {
-    String arg
-  }
-  command {
-    python process.py ${arg}
-  }
+task cpu_example {
+  #....	
   runtime {
-    docker: "ubuntu:latest"
+    cpu: 8
   }
 }
+
 ```
 
-#### memory
+#### memory (**Required**)
 
-Memory requirements for this task.  Two kinds of values are supported for this attributes:
+The `memory` key defines the _minimum_ memory required for this task which must be available prior to the engine starting execution. The engine does not need to provide the exact amount of memory requested, however it may ONLY provision more memory then requested and not less. For example, if the wdl requested `1 GB` but only blocks of `4 GB` were available, the engine might choose to provision `4.0 GB` instead.  Two kinds of values are supported for this attribute:
 
 * `Int` - Interpreted as bytes
 * `String` - This should be a decimal value with suffixes like `B`, `KB`, `MB` or binary suffixes `KiB`, `MiB`.  For example: `6.2 GB`, `5MB`, `2GiB`.
```

Comment:
Should there or should there not be a space between the number and the suffix? Or does it not matter?